perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -993,6 +993,7 @@ private void setRequiredAttributes(PerunSession sess, Member member, Group group
 		for (Member member: membersToRemove) {
 			member.setSourceGroupId(sourceGroupId);
 			groupsManagerImpl.removeMember(sess, group, member);
+			addMemberToGroupsFromTriggerAttribute(sess, group, member);
```

Comment:
I am not sure if this is the right place. You should be sure that removing of the membership is the last connection of member to the group.